src/dotnet/App.Maui/Platforms/Android/IncomingShareHandler.cs

```diff
@@ -0,0 +1,110 @@
+﻿using ActualChat.Chat.UI.Blazor.Services;
+using ActualChat.UI.Blazor.Services;
+using Android.Content;
+using Android.OS;
+using Activity = Android.App.Activity;
+using Uri = Android.Net.Uri;
+
+namespace ActualChat.App.Maui;
+
+public class IncomingShareHandler
+{
+    private ILogger Log { get; set; } = NullLogger.Instance;
+
+    public void OnPostCreate(Activity activity, Bundle? savedInstanceState)
+    {
+        Log = AppServices.LogFor(GetType());
+        TryHandleSend(activity.Intent);
+    }
+
+    public void OnNewIntent(Activity activity, Intent? intent)
+        => TryHandleSend(intent);
+
+    private void TryHandleSend(Intent? intent)
+    {
+        if (intent == null)
+            return;
+        var action = intent.Action;
+        if (!OrdinalEquals(action, Intent.ActionSend) &&
+            !OrdinalEquals(action, Intent.ActionSendMultiple))
+            return;
+
+        var mimeType = intent.Type ?? "";
+        var hasExtraStream = intent.Extras?.ContainsKey(Intent.ExtraStream) ?? false;
+        if (OrdinalEquals(action, Intent.ActionSend)) {
+            if (OrdinalEquals(mimeType, System.Net.Mime.MediaTypeNames.Text.Plain))
+                _ = HandlePlainTextSend(intent.GetStringExtra(Intent.ExtraText));
+            else if (hasExtraStream) {
+                 var stream = Build.VERSION.SdkInt >= BuildVersionCodes.Tiramisu
+                    ? intent.GetParcelableExtra(Intent.ExtraStream, Java.Lang.Class.FromType(typeof(Uri)))
+                    : intent.GetParcelableExtra(Intent.ExtraStream);
+                if (stream is Uri uri)
+                    _ = HandleFilesSend(mimeType, new []{ uri });
+                else
+                    Log.LogWarning("Unsupported stream type: '{StreamType}'", stream?.ToString() ?? "<null>");
+            }
+            else
+                Log.LogWarning("Unsupported send mime type: '{MimiType}'", mimeType);
```

Comment:
Unsupported send mime type: '"application/pdf"'
Please check whether we can share pdf or other attachment documents